sdk/resourcemanager/Azure.ResourceManager.Core/src/Generated/ProviderContainer.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// A class representing collection of resources and their operations over their parent.
+    /// </summary>
+    public class ProviderContainer : ResourceContainerBase<TenantProviderIdentifier, Provider, ProviderData>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProviderContainer"/> class for mocking.
+        /// </summary>
+        protected ProviderContainer()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProviderContainer"/> class.
+        /// </summary>
+        /// <param name="parent"> The client context to use. </param>
+        internal ProviderContainer(OperationsBase parent)
+            : base(parent)
+        {
+        }
+
+        /// <inheritdoc/>
+        protected override ResourceType ValidResourceType => ResourceIdentifier.RootResourceIdentifier.ResourceType;
+
+        /// <inheritdoc/>
+        protected override void Validate(ResourceIdentifier identifier)
+        {
+        }
+
+        private ProviderRestOperations RestClient
+        {
+            get
+            {
+                string subscription;
+                if (Id is null || !Id.TryGetSubscriptionId(out subscription))
+                {
+                    subscription = Guid.Empty.ToString();
+                }
+
+                return new ProviderRestOperations(
+                    Diagnostics,
+                    Pipeline,
+                    subscription,
+                    BaseUri);
+            }
+        }
+
+        /// <summary>
+        /// Gets the provider for a namespace.
+        /// </summary>
+        /// <param name="resourceProviderNamespace"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public Response<Provider> Get(string resourceProviderNamespace, CancellationToken cancellationToken = default)
+        {
+            using var scope = Diagnostics.CreateScope("ProviderOperations.Get");
+            scope.Start();
+
+            try
+            {
+                Response<ProviderData> result;
+                if (Id.GetType().Name == "SubscriptionResourceIdentifier")
```

Comment:
safer to do an is check here
```suggestion
                if (Id is SubscriptionResourceIdentifier)
```